src/EmployerPR/SFA.DAS.EmployerPR/Application/Commands/PostPermissions/PostPermissionsCommandHandler.cs

```diff
@@ -1,12 +1,142 @@
+public class PostPermissionsCommandHandler(IProviderRelationshipsApiRestClient _providerRelationshipsApiRestClient) : IRequestHandler<PostPermissionsCommand, Unit>
 {
-    public async Task<PostPermissionsCommandResult> Handle(PostPermissionsCommand command, CancellationToken cancellationToken)
+    private const int PERMITS_APPROVAL = 1;
+    private const int PERMITS_RECRUITNENT = 1;
+    private const int PERMITS_RECRUITNENT_WITH_REVIEW = 2;
+
+    public async Task<Unit> Handle(PostPermissionsCommand command, CancellationToken cancellationToken)
+    {
+        Response<GetPermissionsResponse> permissionsResponse = 
+            await _providerRelationshipsApiRestClient.GetPermissions(
+                command.Ukprn, 
+                command.AccountLegalEntityId, 
+                cancellationToken
+        );
+
+        List<Operation> existingOperations = [];
+
+        switch(permissionsResponse.ResponseMessage.StatusCode)
+        {
+            case HttpStatusCode.OK:
+                {
+                    existingOperations = permissionsResponse.GetContent().Operations;
+                }
+                break;
+            case HttpStatusCode.NotFound:
+                {
+                    existingOperations = [];
+                }
+                break;
+            default:
+                {
+                    throw new InvalidOperationException($"Invalid operation occurred trying to retrieve permissions for ukprn {command.Ukprn} and AccountLegalEntityId {command.AccountLegalEntityId}");
+                }
+        }
+
+        if (NoUpdatesRequired(command.Operations, existingOperations))
+        {
+            return Unit.Value;
+        }
+
+        PermissionEmailTemplateType templateType = default;
+
+        if (command.Operations.Count == 0)
+        {
+            await _providerRelationshipsApiRestClient.RemovePermissions(
+                command.UserRef, 
+                command.Ukprn!.Value, 
+                command.AccountLegalEntityId, 
+                cancellationToken
+            );
+
+            templateType = PermissionEmailTemplateType.PermissionDeleted;
+        }
+        else
+        {
+            await _providerRelationshipsApiRestClient.PostPermissions(
+                command,
+                cancellationToken
+            );
+
+            templateType = GetTemplateType(existingOperations);
+        }
+
+        await PostNotification(command, templateType, cancellationToken);
+
+        return Unit.Value;
+    }
+
+    private static bool NoUpdatesRequired(List<Operation> requestedPermissions, List<Operation> currentPermissions)
+    {
+        return (requestedPermissions.Count == 0 && currentPermissions.Count == 0) || IdenticalPermissions(requestedPermissions, currentPermissions);
+    }
+
+    private static bool IdenticalPermissions(List<Operation> requestedPermissions, List<Operation> currentPermissions)
```

Comment:
booleans should read like questions `HasIdenticalPermissions` or `IdenticalPermissionsExists`